v0.39-14: flaky: memory hygiene scan tests depend on 1ms wall-clock budgets

## What

`TestMemoryHygieneScan_RevisionChurnReturnsPartialOnlyAfterDurationExhausts` fails intermittently on CI:

```
--- FAIL: TestMemoryHygieneScan_RevisionChurnReturnsPartialOnlyAfterDurationExhausts (0.00s)
    memory_usecase_hygiene_test.go:656: revision-churn Scan() error = memory hygiene continuation
    cannot progress: time_limit budget stopped before advancing the cursor:
    memory hygiene continuation cannot progress
```

Observed on the CI run for PR #1770, whose diff touches only `presentation/cli/` and `docs/` — it cannot have caused a failure in `application/usecase`. Re-running the same job with no code change turned it green, which is the definition of a flake.

## Why it flakes

`application/usecase/memory_usecase_hygiene_test.go:644` builds the budget with `MaxDuration: time.Millisecond`. The test's intent is that the duration budget is exhausted *after* the scan makes progress, so the scan returns a partial result. On a loaded runner, 1 ms can elapse before the first page advances the cursor, so the scan exhausts the budget having made no progress and returns the "cannot progress" error instead of a partial result.

The assertion at `:665` (`result.Usage.Elapsed >= shortBudget.MaxDuration()`) is also wall-clock dependent.

20 consecutive local runs pass, which fits a load-dependent race rather than a logic error.

## Why it matters

A wall-clock threshold this tight makes the test a coin flip on shared runners. Each false red costs a re-run and, worse, trains reviewers to re-run red CI instead of reading it — which is how a real regression gets waved through.

`memory_usecase_hygiene_test.go:785` uses the same `time.Millisecond` budget and likely has the same exposure.

## Proposed fix

Make progress-versus-exhaustion deterministic instead of timing it. The scan already takes its clock from the criteria, so the test can drive a fake clock that advances only at the points the test chooses, and assert the partial result rather than an elapsed duration. Failing that, the "cannot progress" outcome could be made non-fatal when zero rows were scanned — but a fake clock is the honest fix, since the behaviour under test is a state machine, not a stopwatch.

## Done when

- Neither test depends on real elapsed wall-clock time
- The partial-result and cannot-progress branches are each pinned by a deterministic case
- `go test ./application/usecase/ -count=50` is green under CPU load

## Context

Found while reviewing #1687 (PR #1770) in v0.34 Wave 4. Filed as a follow-up rather than widening that PR's scope.

--- a/CHANGELOG.ja.md
+++ b/CHANGELOG.ja.md
@@ -8,6 +8,7 @@ release note と同じ粒度で、版ごとの要点だけをまとめていま�
 ## [Unreleased]
 
 ### Fixed
+- **memory hygiene scan テストが 1ms の wall-clock 予算に依存しない (#1771)** — partial（revision churn）と cannot-progress は fake clock を進め、`MaxDuration` は 1 時間なので `context.WithTimeout` が先に切れない。scratch: 両ケースと `go test ./application/usecase/ -count=50`。
 - **`go test ./presentation/cli/ -race` が export-test seam で data race を出さない (#1698)** — `Set*Func` seam は `atomic.Pointer` スロットなので、serial なテストが差し替えているあいだに parallel なテストが `resolveDBPath` / `userHomeDirFunc` を読んでも定義済みです。production はスロットを書きません。scratch: 並行差し替え対 `resolveDBPath` のテストと、`-race -count=1` を 2 回走らせて `DATA RACE` なし。
 
 ## [v0.38.0] - 2026-08-16

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@ It mirrors the same level of detail as the GitHub release notes, but keeps the h
 ## [Unreleased]
 
 ### Fixed
+- **Memory-hygiene scan tests no longer depend on a 1ms wall-clock budget (#1771)** — the partial (revision-churn) and cannot-progress branches drive a fake clock and use a one-hour `MaxDuration` so `context.WithTimeout` cannot expire first. Scratch: both cases plus `go test ./application/usecase/ -count=50`.
 - **`go test ./presentation/cli/ -race` no longer reports a data race on export-test seams (#1698)** — `Set*Func` seams are `atomic.Pointer` slots, so a serial test can replace them while a parallel test still reads `resolveDBPath` / `userHomeDirFunc`. Production never writes the slots. Scratch: a concurrent replace-vs-`resolveDBPath` test plus two `-race -count=1` package runs report no `DATA RACE`.
 
 ## [v0.38.0] - 2026-08-16

--- a/application/usecase/memory_usecase_hygiene_test.go
+++ b/application/usecase/memory_usecase_hygiene_test.go
@@ -16,6 +16,11 @@ import (
 	domtypes "github.com/duck8823/traceary/domain/types"
 )
 
+// memoryHygieneDeterministicDuration is far larger than any wall-clock the
+// test process can spend, so context.WithTimeout(budget.MaxDuration()) cannot
+// fire first. Exhaustion is driven by the fake scan clock (#1771).
+const memoryHygieneDeterministicDuration = time.Hour
+
 func memoryHygieneTestBudget(
 	t *testing.T,
 	rows int,
@@ -643,7 +648,9 @@ func TestMemoryHygieneScan_RevisionChurnReturnsPartialOnlyAfterDurationExhausts(
 		MaxScanBytes:   1 << 20,
 		MaxResultBytes: 1 << 20,
 		MaxComparisons: 100,
-		MaxDuration:    time.Millisecond,
+		// Larger than any real wall-clock the test can spend; the stub
+		// advances the fake clock by this amount so exhaustion is deterministic (#1771).
+		MaxDuration: memoryHygieneDeterministicDuration,
 	})
 	if err != nil {
 		t.Fatalf("MemoryHygieneScanBudgetFrom() error = %v", err)
@@ -774,7 +781,7 @@ func TestMemoryHygieneScan_TimeLimitWithoutCursorProgressReturnsTypedError(t *te
 	scope := workspaceScope(t, "github.com/example/repo")
 	query := &stubMemoryQueryService{
 		scanClock:   &memoryHygieneTestClock{now: time.Date(2026, 7, 26, 0, 0, 0, 0, time.UTC)},
-		scanAdvance: time.Millisecond,
+		scanAdvance: memoryHygieneDeterministicDuration,
 		summaries: []apptypes.MemorySummary{
 			acceptedSummaryAt(t, "mem-a", scope, "git status", time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC)),
 		},
@@ -785,7 +792,7 @@ func TestMemoryHygieneScan_TimeLimitWithoutCursorProgressReturnsTypedError(t *te
 		MaxScanBytes:   1 << 20,
 		MaxResultBytes: 1 << 20,
 		MaxComparisons: 1,
-		MaxDuration:    time.Millisecond,
+		MaxDuration:    memoryHygieneDeterministicDuration,
 	})
 	if err != nil {
 		t.Fatalf("MemoryHygieneScanBudgetFrom() error = %v", err)

